### PR TITLE
feat/add claimable eth_proxy

### PR DESCRIPTION
**Pull request by fcavazzoli**
#22

### CODE REVIEW

File: eth_bridge/ic/src/eth_proxy/src/api/mint.rs [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+use crate::proxy::{FromNat, ToNat, STATE, TERA_ADDRESS, WETH_ADDRESS_ETH};
+use ic_cdk::export::candid::{Nat, Principal};
+
+use crate::common::types::{
+    IncomingMessageHashParams, Message, MessageStatus, Nonce, TokendId, TxError, TxReceipt,
+};
+
+#[update(name = "mint")]
+#[candid_method(update, rename = "mint")]
+pub async fn mint(token_id: TokendId, nonce: Nonce, payload: Vec<Nat>) -> TxReceipt {
```
Comment (b0xtch) [inline eth_bridge/ic/src/eth_proxy/src/api/mint.rs:16]: There is some stuff missing from this branch `feat/eth_proxy_with_state`

File: eth_bridge/ic/src/eth_proxy/src/api/handle_message.rs [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+            .trim_start_matches("0x")
+            .to_ascii_lowercase())
+    {
+        return Err(TxError::Other(format!(
+            "Eth Contract Address is inccorrect: {}",
+            eth_addr_hex
+        )));
+    }
+
+    mint(eth_addr, nonce, payload).await
```
Comment (b0xtch) [inline eth_bridge/ic/src/eth_proxy/src/api/handle_message.rs:26]: remove `eth_addr`

File: eth_bridge/ic/src/eth_proxy/src/api/mint.rs (hunk 1 above)
Comment (b0xtch) [reply eth_bridge/ic/src/eth_proxy/src/api/mint.rs:16]: shouldn't take `token_id`

File: eth_bridge/ic/src/eth_proxy/src/api/withdraw.rs [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    macros::update,
+    Principal,
+};
+
+/// withdraw left over balance if burn/mint fails
+/// this will attempt to bridge the leftover balance
+/// todo withdraw specific balance
+#[update(name = "withdraw")]
+#[candid_method(update, rename = "withdraw")]
+pub async fn withdraw(token_id: TokendId, eth_addr: EthereumAddr, _amount: Nat) -> TxReceipt {
```
Comment (b0xtch) [inline eth_bridge/ic/src/eth_proxy/src/api/withdraw.rs:21]: Same comment as before, pull from `feat/eth_proxy_with_state` and use hardcode eth address

**Review by b0xtch — APPROVED** [review]
MINT, few comments similar to #19. 

I am a 🧞‍♂️, you must pick between yarn.lock or package-lock your choice 😃

File: eth_bridge/ic/src/eth_proxy/Cargo.toml [hunk 4]
```diff
@@ -1,20 +1,20 @@
...
+edition = "2021"
 
 [lib]
-path = "src/lib.rs"
+path = "src/main.rs"
 crate-type = ["cdylib"]
 
 [dependencies]
+ic-kit = "0.4.4"
+ic-cdk = "0.4.0"
```
[the remark this answers is not in the document]
Comment (fcavazzoli) [reply eth_bridge/ic/src/eth_proxy/Cargo.toml:12]: updated, also updated the version in weth canister it was 0.3.1

**Pull request merged**